datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/UpdateParentNodeResolver.java

```diff
@@ -41,20 +41,25 @@ public CompletableFuture<Boolean> get(DataFetchingEnvironment environment) throw
       throw new IllegalArgumentException(String.format("Failed to update %s. %s does not exist.", targetUrn, targetUrn));
     }
 
-    GlossaryNodeUrn parentNodeUrn = GlossaryNodeUrn.createFromString(input.getParentNode());
-    if (!_entityService.exists(parentNodeUrn) || !parentNodeUrn.getEntityType().equals(Constants.GLOSSARY_NODE_ENTITY_NAME)) {
-      throw new IllegalArgumentException(String.format("Failed to update %s. %s either does not exist or is not a glossaryNode.", targetUrn, parentNodeUrn));
+    GlossaryNodeUrn parentNodeUrn = null;
+    if (input.getParentNode() != null) {
+      parentNodeUrn = GlossaryNodeUrn.createFromString(input.getParentNode());
+      if (!_entityService.exists(parentNodeUrn) || !parentNodeUrn.getEntityType().equals(Constants.GLOSSARY_NODE_ENTITY_NAME)) {
+        throw new IllegalArgumentException(String.format("Failed to update %s. %s either does not exist or is not a glossaryNode.", targetUrn, parentNodeUrn));
+      }
     }
+
+    GlossaryNodeUrn finalParentNodeUrn = parentNodeUrn;
```

Comment:
minor: is this reassignment necessary here?

Reply:
Yup, IntelliJ is throwing this lint error 🥲  : `Variable used in lambda expression should be final or effectively final`. This was the proposed solution provided